samples/go/src/auth/auth.go

```diff
@@ -0,0 +1,48 @@
+package auth
+
+import (
+	"net/http"
+	"os"
+
+	"leadIntegration/src/utils"
+)
+
+func ValidAuthorization(user string, secretKey string) bool {
+
+	if secretKey == "" ||
+		user == "" ||
+		secretKey != os.Getenv("SECRET_KEY") ||
+		user != "vivareal" {
+		return false
+	}
+
+	return true
+}
+
+var BasicAuthentication = func(next http.Handler) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
this middleware is strange, as this sample has only 2 routes, why not implement it directly in the controller that needs basic auth 🤔

Reply:
because I wanted to separate authorization, so if any other route need to be implemented it will use this middleware

Reply:
but, will you implement some more routes in this sample code?

Reply:
no, only these.

I changed it, now this verification is done directly in the controller